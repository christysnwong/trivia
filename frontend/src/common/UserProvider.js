import React, {useEffect, useState} from "react";
import UserContext from "./UserContext";

const demoUser = {
  id: 1,
  username: "testuser",
  first_name: "Test",
  last_name: "User",
  email: "test@test.com",
  stats: {
    level: 14,
    title: "Pro",
    "quizzes completed": 34,
    points: 3440,
    remainingPts: 60,
    levelPts: 300,
  },
};

const UserProvider = ({
  children,
  currUser = demoUser
}) => {
  
  const [currentUser, setCurrUser] = useState();

  return (
    // <UserContext.Provider value={{ currUser }}>
    <UserContext.Provider value={{ currUser, setCurrUser }}>
      {children}
    </UserContext.Provider>
  );
};

export default UserProvider;
