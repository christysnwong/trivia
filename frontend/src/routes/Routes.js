
import { Route, Switch, Redirect } from "react-router-dom";

import Home from "../home/Home.js";

import Quiz from "../quizzes/Quiz"
import Leaderboard from "../userInfo/Leaderboard"


import LoginForm from "../usersForms/LoginForm";
import SignUpForm from "../usersForms/SignUpForm";
import ProfileEditForm from "../usersForms/ProfileEditForm";

import Profile from "../userInfo/Profile";
import Stats from "../userInfo/Stats";
import PersonalBest from "../userInfo/PersonalBest";
import Sessions from "../userInfo/Sessions";

import PrivateRoute from "./PrivateRoute";

import Favourite from "../userFavs/Favourite.js";



const Routes = ({ login, signup }) => {

  // console.debug(
  //   "Routes",
  //   `login=${typeof login}`,
  //   // `register=${typeof register}`
  // );

  return (
    <div className="pt-5">
      <Switch>
        <Route exact path="/">
          <Home />
        </Route>

        <Route exact path="/quizzes">
          <Quiz />
        </Route>

        <Route exact path="/leaderboard">
          <Leaderboard />
        </Route>

        <Route exact path="/favourites">
          <Favourite />
        </Route>

        <PrivateRoute exact path="/profile">
          <Profile />
        </PrivateRoute>

        <PrivateRoute exact path="/profile/edit">
          <ProfileEditForm />
        </PrivateRoute>

        <PrivateRoute exact path="/stats">
          <Stats />
        </PrivateRoute>

        <PrivateRoute exact path="/personalbest">
          <PersonalBest />
        </PrivateRoute>

        <PrivateRoute exact path="/sessions">
          <Sessions />
        </PrivateRoute>

        <Route exact path="/login">
          <LoginForm login={login} />
        </Route>
        <Route exact path="/signup">
          <SignUpForm signup={signup} />
        </Route>
        <Route exact path="/logout">
          <Redirect to="/" />
        </Route>

        <Redirect to="/" />
      </Switch>
    </div>
  );
};


export default Routes;
