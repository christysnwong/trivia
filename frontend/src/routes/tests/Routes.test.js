import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import Routes from "../Routes";
import UserProvider from "../../common/UserProvider";

it("renders without crashing", function () {
  render(
    <MemoryRouter>
      <UserProvider>
        <Routes />
      </UserProvider>
    </MemoryRouter>
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <UserProvider>
        <Routes />
      </UserProvider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});