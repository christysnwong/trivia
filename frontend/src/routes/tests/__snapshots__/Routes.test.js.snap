// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="pt-5"
  >
    <div
      class="col-10 offset-1 col-lg-8 offset-lg-2 col-xxl-6 offset-xxl-3 col-xxxl-4 offset-xxxl-4"
    >
      <div
        class="text-center shadow-1-strong rounded mb-2 logo-bg"
      >
        <div
          class="p-4 w-100 mask-bg"
        >
          <h5
            class="mb-3"
          >
            Welcome back to 
          </h5>
          <h1
            class="display-4 brand"
          >
            Trivia Guru
          </h1>
          <hr
            class="mb-2"
          />
          <p
            class="lead fw-bold"
          >
            Test your knowledge and learn more everyday!
          </p>
        </div>
      </div>
      <div
        class="row mb-5"
      >
        <div
          class="col-12"
        >
          <h3
            class="mb-3 mt-3"
          >
            <i
              class="fa-solid fa-id-badge me-2"
            />
            Recent Activities
          </h3>
        </div>
        <div
          class="col-8"
        >
          <p>
            Hi testuser! You are currently a 
            <b>
              Level 14 Pro
            </b>
            .
            <br />
          </p>
          <p>
            You are 60 points away from the next level.
          </p>
          <div
            class="progress"
          >
            <div
              aria-valuemax="300"
              aria-valuemin="0"
              aria-valuenow="60"
              class="progress-bar bg-warning"
              role="progressbar"
              style="width: 80%;"
            />
          </div>
        </div>
        <div
          class="col-4"
        >
          <div
            class="mt-2 float-end"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
